samza-core/src/main/java/org/apache/samza/serializers/model/SamzaObjectMapper.java

```diff
@@ -89,10 +91,30 @@ public static ObjectMapper getObjectMapper() {
     mapper.getSerializationConfig().addMixInAnnotations(TaskModel.class, JsonTaskModelMixIn.class);
     mapper.getDeserializationConfig().addMixInAnnotations(TaskModel.class, JsonTaskModelMixIn.class);
     mapper.getSerializationConfig().addMixInAnnotations(ContainerModel.class, JsonContainerModelMixIn.class);
-    mapper.getDeserializationConfig().addMixInAnnotations(ContainerModel.class, JsonContainerModelMixIn.class);
     mapper.getSerializationConfig().addMixInAnnotations(JobModel.class, JsonJobModelMixIn.class);
     mapper.getDeserializationConfig().addMixInAnnotations(JobModel.class, JsonJobModelMixIn.class);
 
+    module.addDeserializer(ContainerModel.class, new JsonDeserializer<ContainerModel>() {
```

Comment:
Minor: Slightly prefer being consistent with all other `DeSerializer`s. All others appear to define static inner classes as opposed to inlining them.

`  public static class ConfigDeserializer extends JsonDeserializer<Config> `